elrond-wasm-derive/src/parse_attr/endpoint_attr.rs

```diff
@@ -0,0 +1,126 @@
+use super::attr_names::*;
+use super::util::*;
+
+pub fn is_callback_decl(m: &syn::TraitItemMethod) -> bool {
+	has_attribute(&m.attrs, ATTR_CALLBACK_DECL)
+}
+
+pub fn is_callback_raw_decl(m: &syn::TraitItemMethod) -> bool {
+	has_attribute(&m.attrs, ATTR_CALLBACK_RAW_DECL)
+}
+
+pub fn is_init(m: &syn::TraitItemMethod) -> bool {
+	has_attribute(&m.attrs, ATTR_INIT)
+}
+
+pub fn is_var_args(pat: &syn::PatType) -> bool {
+	has_attribute(&pat.attrs, ATTR_VAR_ARGS)
+}
+
+pub fn is_callback_arg(pat: &syn::PatType) -> bool {
+	has_attribute(&pat.attrs, ATTR_CALLBACK_ARG)
+}
+
+#[derive(Clone, Debug)]
+pub struct EndpointAttribute {
+	pub endpoint_name: Option<syn::Ident>,
+}
+
+impl EndpointAttribute {
+	pub fn parse(m: &syn::TraitItemMethod) -> Option<EndpointAttribute> {
+		match find_attr_with_one_opt_token_tree_arg(m, ATTR_ENDPOINT) {
+			None => None,
+			Some(Some(proc_macro2::TokenTree::Ident(ident))) => Some(EndpointAttribute {
+				endpoint_name: Some(ident),
+			}),
+			Some(None) => Some(EndpointAttribute {
+				endpoint_name: None,
+			}),
+			_ => panic!("unexpected endpoint argument tokens"),
+		}
+	}
+}
+
+#[derive(Clone, Debug)]
+pub struct ViewAttribute {
+	pub view_name: Option<syn::Ident>,
+}
+
+impl ViewAttribute {
+	pub fn parse(m: &syn::TraitItemMethod) -> Option<ViewAttribute> {
+		match find_attr_with_one_opt_token_tree_arg(m, ATTR_VIEW) {
+			None => None,
+			Some(Some(proc_macro2::TokenTree::Ident(ident))) => Some(ViewAttribute {
+				view_name: Some(ident),
+			}),
+			Some(None) => Some(ViewAttribute { view_name: None }),
+			_ => panic!("unexpected endpoint argument tokens"),
```

Comment:
Having to write `Some(Some(something))` suggests that `find_attr_with_one_opt_token_tree_arg()` should be returning some sort of `Enum`. E.g. an `Enum` containing the values `NoAttribute`, `NullaryAttribute { name-or-something }` instead of `None` and `Some(None)` respectively.